src/components/external-source/ExternalSourceManager.svelte

```diff
@@ -0,0 +1,1237 @@
+    externalSourceType.name.toLowerCase().includes(filterString),
+  );
+  $: filteredTableExternalEvents = selectedEvents.filter(event => {
+    const filterTextLowerCase = externalEventsTableFilterString.toLowerCase();
+    const includesName = externalEventsTableFilterString.length
+      ? event.pkey.key.toLocaleLowerCase().includes(filterTextLowerCase)
+      : true;
+    return includesName;
+  });
+  $: effects.getExternalEvents(selectedSource ? selectedSource.pkey : null, user).then(
+    fetched =>
+      (selectedEvents = fetched.map(eDB => {
+        return {
+          ...eDB,
+          duration_ms: convertDurationToMs(eDB.duration),
+          event_type: eDB.pkey.event_type_name,
+          start_ms: convertUTCtoMs(eDB.start_time),
+        };
+      })),
+  );
+  $: selectedSourceLinkedDerivationGroupsPlans = $planDerivationGroupLinks.filter(planDerivationGroupLink => {
+    return planDerivationGroupLink.derivation_group_name === selectedSource?.pkey.derivation_group_name;
+  });
+
+  // Permissions
+  $: hasDeletePermission = featurePermissions.externalSource.canDelete(user);
+  $: hasCreatePermission = featurePermissions.externalSource.canCreate(user);
+
+  async function onDeleteExternalSource(selectedSource: ExternalSourceSlim | null | undefined) {
+    if (selectedSource !== null && selectedSource !== undefined) {
+      // selectedSource here does not necessarily align with global selected source, especially if you click delete
+      //    in a table without making a selection. as such, can't use selectedSourceLinkedDerivationGroupsPlans, must
+      //    make a new one.
+      let currentlyLinked = $planDerivationGroupLinks.filter(planDerivationGroupLink => {
+        return planDerivationGroupLink.derivation_group_name === selectedSource?.pkey.derivation_group_name;
+      });
+      if (currentlyLinked.length > 0) {
+        // if the source is in a derivation group currently used by a plan, warn that we cannot delete
+        await showDeleteExternalSourceModal(currentlyLinked, selectedSource);
+      } else {
+        // otherwise, delete!
+        const deletionWasSuccessful = await effects.deleteExternalSource(selectedSource, user);
+        if (deletionWasSuccessful) {
+          deselectSource();
+        }
+      }
+    }
+  }
+
+  async function onFormSubmit(_e: SubmitEvent) {
+    if (parsed && file) {
+      // Setup insert inputs for Hasura mutations
+      // External Source Type
+      sourceTypeInsert = {
+        name: $sourceTypeField.value,
+      };
+
+      // Derivation Group
+      derivationGroupInsert = {
+        name:
+          $derivationGroupField.value.length !== 0 ? $derivationGroupField.value : `${sourceTypeInsert.name} Default`,
+        source_type_name: sourceTypeInsert.name,
+      };
+
+      // External Source
+      const start_time: string | undefined = convertDoyToYmd($startTimeDoyField.value.replaceAll('Z', ''))?.replace(
```

Comment:
This conversion is not about what the user faces but rather what SQL accepts as valid input. The unformatted strings would fail when pushed to SQL, hence this conversion. All dates in the UI are presented in DOY format.